src/ExactDiagonalization/operator_as_map.jl

```diff
@@ -0,0 +1,123 @@
+"""
+    OperatorAsMap <: LinearMap
+
+Wrapper over [`AbstractOperator`](@ref) that allows it to be used as a [`LinearMap`](@ref) from [LinearMaps.jl](https://github.com/JuliaLinearAlgebra/LinearMaps.jl).
+
+See [`LinearMap`](@ref) for usage.
```

Comment:
Would it make sense to define methods for `BasisSetRepresentation`, `sparse` and `Matrix` for `OperatorAsMap` as a way to instantiate the exact same matrix as defined by the map?